### PR TITLE
feat(cli): make all stake arguments required and explicitly check validator correctness

### PR DESCRIPTION
**Pull request by drcpu-github**

### CODE REVIEW

File: node/src/actors/json_rpc/api.rs [hunk 1]
```diff
@@ -1948,62 +1948,24 @@ pub async fn snapshot_import(params: Result<SnapshotImportParams, Error>) -> Jso
...
-            "[STAKE] No withdrawer address was provided, using the node's own address: {}",
-            address
-        );
-
-        address
-    };
+    let withdrawer = PublicKeyHash::from_bech32(get_environment(), &params.withdrawer)
+        .map_err(internal_error)?;
+    log::debug!(
+        "[STAKE] A withdrawer address was provided: {}",
```
Comment (aesedepece) [inline node/src/actors/json_rpc/api.rs:1955]:
```suggestion
        "[STAKE] Creating stake transaction with withdrawer address: {}",
```

File: node/src/actors/messages.rs [hunk 2]
```diff
@@ -248,13 +248,13 @@ impl Message for BuildStake {
 pub struct BuildStakeParams {
     /// Authorization signature and public key
     #[serde(default)]
-    pub authorization: Option<MagicEither<String, KeyedSignature>>,
+    pub authorization: String,
```
Comment (aesedepece) [inline node/src/actors/messages.rs:251]:
```suggestion
    pub authorization: MagicEither<String, KeyedSignature>,
```

File: node/src/actors/messages.rs [hunk 3]
```diff
@@ -248,13 +248,13 @@ impl Message for BuildStake {
...
     /// List of `ValueTransferOutput`s
     #[serde(default)]
     pub value: u64,
     /// Withdrawer
     #[serde(default)]
-    pub withdrawer: Option<MagicEither<String, PublicKeyHash>>,
+    pub withdrawer: String,
```
Comment (aesedepece) [inline node/src/actors/messages.rs:257]:
```suggestion
    pub withdrawer: MagicEither<String, PublicKeyHash>,
```

File: src/cli/node/json_rpc_client.rs [hunk 4]
```diff
@@ -867,8 +867,9 @@ pub fn send_dr(
 pub fn send_st(
     addr: SocketAddr,
     value: u64,
-    authorization: Option<MagicEither<String, KeyedSignature>>,
-    withdrawer: Option<MagicEither<String, PublicKeyHash>>,
+    authorization: String,
```
Comment (aesedepece) [inline src/cli/node/json_rpc_client.rs:870]:
```suggestion
    authorization: MagicEither<String, KeyedSignature>,
```

File: src/cli/node/json_rpc_client.rs [hunk 5]
```diff
@@ -867,8 +867,9 @@ pub fn send_dr(
...
+    validator: String,
```
Comment (aesedepece) [inline src/cli/node/json_rpc_client.rs:871]:
```suggestion
    validator: MagicEither<String, PublicKeyHash>,
```

File: src/cli/node/json_rpc_client.rs [hunk 6]
```diff
@@ -867,8 +867,9 @@ pub fn send_dr(
...
+    withdrawer: String,
```
Comment (aesedepece) [inline src/cli/node/json_rpc_client.rs:872]:
```suggestion
    withdrawer: MagicEither<String, PublicKeyHash>,
```

File: src/cli/node/with_node.rs [hunk 7]
```diff
@@ -273,15 +273,17 @@ pub fn exec_cmd(
...
             require_confirmation,
             dry_run,
         } => rpc::send_st(
             node.unwrap_or(default_jsonrpc),
             value,
-            authorization.map(MagicEither::Left),
-            withdrawer.map(MagicEither::Left),
+            authorization,
+            validator,
+            withdrawer,
```
Comment (aesedepece) [inline src/cli/node/with_node.rs:286]:
```suggestion
            authorization.map(MagicEither::Left),
            validator.map(MagicEither::Left),
            withdrawer.map(MagicEither::Left),
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

**Comment by drcpu-github** [comment]
@aesedepece Not quite sure the `Default` implementation I had to create for `MagicEither` makes a lot of sense, so a renewed review would be appreciated.

**Review by aesedepece — APPROVED** [review]

**Comment by aesedepece** [comment]
> @aesedepece Not quite sure the `Default` implementation I had to create for `MagicEither` makes a lot of sense, so a renewed review would be appreciated.

It's a lesser evil :sweat_smile:

**Pull request merged**